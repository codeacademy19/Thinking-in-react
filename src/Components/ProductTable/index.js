import React, { Component } from 'react';
import ProductCategoryRow from '../ProductCategoryRow';
import ProductRow from '../ProductRow';
const getProductRows=require('../../utils/ProductTable/getProductRows');
class ProductTable extends Component {
    render() {
      const filterText = this.props.filterText;
      const inStockOnly = this.props.inStockOnly;
  
      
      let lastCategory = null;
      const rowsForTable=[];
      const rows = getProductRows(this.props.product,filterText,inStockOnly,lastCategory);
      //console.log(rows);
      rows.forEach((row)=>{
        //console.log(row);
          if(row[0]==='category'){
            rowsForTable.push(<ProductCategoryRow
            category={row[1]}
            key={row[1]} />)
          }else{
            rowsForTable.push(<ProductRow
                product={row[1]}
                key={row[1].name} />)  
          }
      })
  
      return (
        <table>
          <thead>
            <tr>
              <th>Name</th>
              <th>Price</th>
            </tr>
          </thead>
          <tbody>{rowsForTable}</tbody>
        </table>
      );
    }
  }
  export default ProductTable;