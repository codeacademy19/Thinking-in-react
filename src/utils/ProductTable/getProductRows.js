module.exports=(products,filterText,inStockOnly,lastCategory)=>{
    const rows=[];
    products.forEach((product) => {
        if (product.name.indexOf(filterText) === -1) {
           return;
        }
        if (inStockOnly && !product.stocked) {
          return;
        }
        if (product.category !== lastCategory) {
          rows.push(['category',product.category]);
        }
            rows.push(['product',product]);
            lastCategory = product.category;
        
      });
      return rows;
}
