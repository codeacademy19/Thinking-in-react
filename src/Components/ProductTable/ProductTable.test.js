import React from 'react';
import renderer from 'react-test-renderer';
import ProductTable from '.';
import {shallow} from 'enzyme';


describe('ProductRow',()=>{
    const PRODUCTS = [
        {category: 'Sporting Goods', price: '$49.99', stocked: true, name: 'Football'},
        {category: 'Sporting Goods', price: '$9.99', stocked: true, name: 'Baseball'},
        {category: 'Sporting Goods', price: '$29.99', stocked: false, name: 'Basketball'},
        {category: 'Electronics', price: '$99.99', stocked: true, name: 'iPod Touch'},
        {category: 'Electronics', price: '$399.99', stocked: false, name: 'iPhone 5'},
        {category: 'Electronics', price: '$199.99', stocked: true, name: 'Nexus 7'}
      ];
    it('should render a row with value as Product Category',()=>{
        const tree=renderer.create(<ProductTable product={PRODUCTS}  />).toJSON();
        expect(tree).toMatchSnapshot();
    })
    
});