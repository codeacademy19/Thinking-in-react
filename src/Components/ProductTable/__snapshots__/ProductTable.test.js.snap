// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductRow should render a row with value as Product Category 1`] = `
<table>
  <thead>
    <tr>
      <th>
        Name
      </th>
      <th>
        Price
      </th>
    </tr>
  </thead>
  <tbody />
</table>
`;
